### DISCUSSION TITLE
How to access cmdline flags with dashes?

### DISCUSSION THREAD

**Discussion by robaca** [nushell/nushell#7223] [Q&A]
Hi! I want to use flags with dashes in them in my script. How can I access them inside my custom command?
 
```
def main [
   --dry-run(-d) = false 
] {
   # $dry-run or $d are both not working here, how do I access it?
}
```

**Comment by robaca** [comment]
Ok, found the answer by myself - the dashes are replaced by underscores (`$dry_run`).

**Comment by theAkito** [comment]
It'd probably help to apply the same logic from commands with dashes to this problem. I.e. something like this.

```nu
def main [
   "--dry-run" (-d) = false 
] {
  echo $dry-run # This would still be problematic.
}
```

I guess, there is no good solution in the current context of how Nu works.
On the other hand, options with dashes are no the best idea in the first place.
It'd probably be ideal, if the original CLI would just fix the naming.
